### DISCUSSION TITLE
Can `update_app_identifier` be used with an Extension?

### DISCUSSION THREAD

**Discussion by liamjones** [fastlane/fastlane#20384] [Q&A]
This seems impossible from what I've tried so far but maybe I'm missing something...

We have our app's bundle identifier as `xxx.dev`, we produce builds for different environments so have fastlane use `update_app_identifier` to change this to `xxx.test`, etc as appropriate when running a lane.

I'm currently integrating push notifications and the Notification Service Extension needs a similar update. E.g. it's `xxx.dev.extensionId` and would need to change to `xxx.test.extensionId`, etc.

I've tried using the `update_app_identifier` action, pointing it at the extension's `Info.plist` but, whereas it _should_ be changing the `PRODUCT_BUNDLE_IDENTIFIER` inside `xxx.xcodeproj` it instead adds a `CFBundleIdentifier` to the `Info.plist` (which doesn't seem to work for an extension).

Any ideas? I've had a search but everything I've found seems to be about problems with code signing for extensions, not for changing their identifier.

**Comment by joetannenbaum** [comment]
I know this is a bit old, but if anyone else has this question, this is what worked for me. Make sure you have the following in your extension's `Info.plist`:

```
<key>CFBundleIdentifier</key>
<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
```

Without it, `update_app_identifier` will add the `CFBundleIdentifier` but XCode won't acknowledge it.
